### DISCUSSION TITLE
unable to configure OpenDDS-3.21 build on Windows 10

### DISCUSSION THREAD

**Discussion by h3gui** [OpenDDS/OpenDDS#3662] [Q&A] [answered]
Hi, 
I am experiencing issues configuring the OpenDDS-3.21 build.
I used the configure command in Visual Studio Code and I had the error as in the attached file.
The most recent Strawberry Perl is installed on my PC as recommended.
Your help is greatly appreciated. 
Thanks
![configure_error](https://user-images.githubusercontent.com/110626246/182913967-b8ca1dff-b378-4a3e-b08b-053abec13177.png)

**Comment by jrw972** [comment, marked as answer]
Use Visual Studio command prompt as described in the quick start.

**Reply by h3gui** [reply]
Thank you for the prompt reply. 
configure by itself was not sufficient. I used --compiler=VAL but resulted in an error as attached.
![invalid_type_error](https://user-images.githubusercontent.com/110626246/182942638-581c5799-2e85-4cfa-8592-fd276f090854.png)

**Reply by mitza-oci** [reply]
The `VAL` part isn't meant to be literal text, it's what one could use to specify a compiler.  As in `--compiler=\Path\to\some\program`.

But that shouldn't be necessary since `cl` (the executable for the Microsoft Visual C++ compiler) should be already on the PATH.

**Reply by mitza-oci** [reply]
Also, any time the `configure` script is doing something unexpected it's good to run it with `--verbose` and closely examine the log.  Even without the `--verbose` flag, in this case we see the Perl interpreter errors "Use of uninitalized value..." indicating that something has gone wrong.

**Reply by h3gui** [reply]
Thanks Adam, the compiler was not recognized. Any help is appreciated.  Where to find the log file?
![compiler_not_recognized](https://user-images.githubusercontent.com/110626246/183079050-16adb1f7-520f-42c0-8241-66a13b388da2.png)

**Reply by simpsont-oci** [reply]
I think you meant to type `configure --verbose`.

**Reply by h3gui** [reply]
yes, I took the incorrect capture. The compiler was not found. Where to find the log file?
![compiler_not_found](https://user-images.githubusercontent.com/110626246/183086640-8c375081-5882-45b7-b72b-d0f7d8335172.png)

**Reply by mitza-oci** [reply]
Make sure you install the C++ development tools in Visual Studio

**Reply by h3gui** [reply]
Thank you. This solved my issues
